types/tx.go

```diff
@@ -25,11 +25,14 @@ type (
 	TxKey [TxKeySize]byte
 )
 
-// Hash computes the TMHASH hash of the wire encoded transaction.
-func (tx Tx) Hash() []byte {
-	return tmhash.Sum(tx)
+// Hash computes the hash of the transaction.
+//
+// See txhash.Sum.
+func (tx Tx) Hash() txhash.Bytes {
+	return txhash.Sum(tx)
 }
 
+// Key returns the sha256 hash of the transaction.
 func (tx Tx) Key() TxKey {
 	return sha256.Sum256(tx)
```

Comment:
Why aren't we changing this one to use the hash the user has set?

Reply:
Because ` TxKey is the fixed-length array key used as an index, ' we only use it internally, afaik, in the mempool reactor. Do you think we should be using `txhash` here as well?

Reply:
Shouldn't the hash be the key?